src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs

```diff
@@ -346,11 +347,15 @@ private ValueGetter<VBuffer<TValue>> GetGetterCore<TValue>(DataViewRow input, in
                             return;
                         }
 
-                        Host.Check(src.PixelFormat == System.Drawing.Imaging.PixelFormat.Format32bppArgb
-                            || src.PixelFormat == System.Drawing.Imaging.PixelFormat.Format24bppRgb,
-                            "Transform only supports pixel formats Format24bppRgb and Format32bppArgb");
                         Host.Check(src.Height == height && src.Width == width);
 
+                        if (src.PixelFormat != PixelFormat.Format32bppArgb && src.PixelFormat != PixelFormat.Format24bppRgb)
+                        {
+                            var clone = src.Clone(new Rectangle(0, 0, src.Width, src.Height), PixelFormat.Format32bppArgb);
+                            src.Dispose();
```

Comment:
If these need to be disposed, who disposes this clone?

Reply:
@eerhardt https://github.com/dotnet/machinelearning/blob/855a7faeb84230b01a67a72e71b4d95f8d3e3bf7/src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs#L328